### ISSUE TITLE
Trips Sometimes Start Late

### ISSUE THREAD

**Issue by meatberger** [rivountain/ME-Bike#4]
## Description
Sometimes there is a delay between when the user begins riding the bike and when the route starts. If this happens to you, please comment with details about what state the bike, phone, and BLE were in when this glitch happened. Please provide a picture of the BLE and USB adapter (if any) and list any other bluetooth devices that the phone is connected to. This issue was first described here:

![MEBikeIssues](https://user-images.githubusercontent.com/17072467/92493379-70677800-f1c2-11ea-816c-432ea5476e60.png)

**Comment by mojiazad** [comment]
Mojdeh here. Yes, it has been happening to me. The phone isn't connected to any other devices and is located in the backpack. For some reason, this happens to me more when I'm leaving home to school (am/pm does not matter), and not vice versa (it did happen maybe once or twice from school to home, but for home to school there are so many of them). My trips are short so by the time I get to school, it would be only a couple of points. I'm using ME-Bike-2.1.0, and the iPhone is only for this purpose- not my main phone and it's always in the backpack before and during rides. The battery of the e-bike was more than 75% ish full today, not connected to the charger for a couple of days (so no "charging" data or engagement with the app), and lost the whole trip to school except for some random point-where I didn't bike close to them

**Comment by meatberger** [comment]
> Mojdeh here. Yes, it has been happening to me. The phone isn't connected to any other devices and is located in the backpack. For some reason, this happens to me more when I'm leaving home to school (am/pm does not matter), and not vice versa (it did happen maybe once or twice from school to home, but for home to school there are so many of them). My trips are short so by the time I get to school, it would be only a couple of points. I'm using ME-Bike-2.1.0, and the iPhone is only for this purpose- not my main phone and it's always in the backpack before and during rides. The battery of the e-bike was more than 75% ish full today, not connected to the charger for a couple of days (so no "charging" data or engagement with the app), and lost the whole trip to school except for some random point-where I didn't bike close to them

Thank you for this information, @mojiazad ! This really helps a lot in debugging this. I will look into this issue more once I close #1 . I have a couple more questions about this. Are there any other devices around at home that emit BLE? Do you use a smartwatch or AirPods, fitness tracker such as a Fitbit? These could also interfere with BLE connectivity.

**Comment by meatberger** [comment]
Two other issues may not fully resolve this issue, but could alleviate the delay.  #5 will allow for a workaround to BLE triggers in general. I will also be interested to see if refactoring the backend will resolve some delay issues on its own once #1 is finished. With that being said I certainly want to fully resolve this issue as well.

**Comment by mojiazad** [comment]
@meatberger, thank you. No, I have no other devices at all- smartwatch or airpods. I only sometimes wear a Fitbit which is connected to another phone (and the Bluetooth on that phone is usually off).

**Comment by meatberger** [comment]
Great, thank you @mojiazad . In theory, the FitBit's constant BLE could still interfere with the trip's start, even if not connected to any phones. Can you confirm that the issue still happens when the Fitbit is not even nearby? I am wondering if the constant BLE emission is interfering with ME-Bike "noticing" the BLE dongle.

**Comment by mojiazad** [comment]
@meatberger , I will test more and let you know. Another point is that I stay in the same status in terms of carrying a phone(s) and Fitbit (if I do) throughout the day, however, the delay happens mostly starting from home rather than school. I will check other participants' data and let you know if the same happens at their home/work/other stop locations.

**Comment by meatberger** [comment]
Alright, Thanks again @mojiazad . Meanwhile, I will still see if I can reproduce this issue with using an Apple Watch or without Any other BLE devices nearby

**Comment by meatberger** [comment]
Update: I was able to reproduce this issue using an Apple Watch. The watch was not connected to my phone, and did not interfere until I started interacting with bluetooth features on the watch. I kept restarting the app while turning on bluetooth settings on the watch, and eventually got to a state where the trip never started. It stayed this way until I powered the watch down, and then the trip resumed.

**Comment by meatberger** [comment]
It appears the watch was syncing to "Nordic_HRM," and this was causing the application not to start the trip. I can reproduce this by syncing my phone to "Nordic_HRM" (which is the BLE dongle) in settings. The trip will not start until I forget the device in settings. This could be just one case of this issue. Perhaps other users have encountered this without syncing directly to the dongle, but it will still be helpful to see what devices are in range when the issue is encountered. While pairing directly to the BLE is an optional security measure, it is possible that we do not handle pairing directly to the device. This pairing is treated like an OEM bluetooth connection at this point, so this case of producing the issue could be resolved by #5.

TL;DR; syncing directly to the dongle is definitely one cause of this issue.

**Comment by meatberger** [comment]
@mojiazad can you confirm whether the iPhone you use for testing is synced directly to the BLE in your Bluetooth settings? If so, that may be causing the delay.

**Comment by meatberger** [comment]
When consider possible causes for this — should it present itself post-refactor — please consider the permissions to constantly update location:

![Research Title E-bike user behavior and sust copy](https://user-images.githubusercontent.com/17072467/95784394-b8862880-0ca1-11eb-82a8-72916820b2e4.png)
